clarifai.go

```diff
@@ -0,0 +1,245 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"sort"
+	"strings"
+)
+
+const (
+	clarifaiApi = "https://api.clarifai.com/v2/models/aaa03c23b3724a16a56b629203edc62c/outputs"
+	clarifyAuth = "https://api.clarifai.com/v1/token"
+	minValue    = 0.72
+	maxTags     = 10
+)
+
+type ClarifaiClient struct {
+	Token  string
+	Expiry int
+	Access bool
+}
+
+type ClarifyTag struct {
+	Name  string  `json:"name"`
+	Value float64 `json:"value"`
+}
+
+func NewClarifaiClient() *ClarifaiClient {
+	return &ClarifaiClient{
+		Access: false,
+	}
+}
+
+// Check whether the client still has access rights to Clarifai
+func (client *ClarifaiClient) isAccess() bool {
+	if !client.Access || client.Expiry <= 0 {
+		client.Access = false
+		return false
+	}
+	return true
+}
+
+// Refresh the access token
+func (client *ClarifaiClient) RefreshAccesToken() {
+	CLARIFAI_ID := os.Getenv("CLARIFAI_CLIENT_ID")
+	CLARIFAI_SECRET_KEY := os.Getenv("CLARIFAI_SECRET_KEY")
+
+	responseData := authRequest(CLARIFAI_ID, CLARIFAI_SECRET_KEY)
+	var token struct {
+		//type AccessToken struct {
+		Token  string `json:"access_token"`
+		Expiry int    `json:"expires_in"`
+		//}
+	}
+	rawIn := json.RawMessage(responseData)
+
+	b, err := rawIn.MarshalJSON()
+	if err != nil {
+		fmt.Println("ERROR Parsing authentication response")
+		panic(err)
+	}
+
+	err = json.Unmarshal(b, &token)
+	if err != nil {
+		fmt.Println("ERROR Parsing authentication json string")
+		panic(err)
+	}
+
+	fmt.Printf("Token: %v\n", token.Token)
+	fmt.Printf("Expiry: %v\n", token.Expiry)
+
+	client.Token = token.Token
+	client.Expiry = token.Expiry
+	client.Access = true
+
+}
+
+func authRequest(CLIENT_ID string, SECRET_KEY string) []byte {
+	resp, err := http.PostForm(clarifyAuth, url.Values{"grant_type": {"client_credentials"},
+		"client_id":     {CLIENT_ID},
+		"client_secret": {SECRET_KEY}})
+
+	fmt.Print("\nAuthentication Request constructed and sent")
+
+	if err != nil {
+		panic(err)
```

Comment:
Yeah, I didn't really touch the existing code that wasn't part of the refactoring. We can do that later (we have some issue tickets open for that)